test/helpers/comparison-utils.js

```diff
@@ -265,31 +338,44 @@ export function cmpParsedAppConfigs(t, expectedConfig, actualConfig, {name} = {}
   });
 }
 
+export function cmpFields(t, expected, actual, name) {
+  t.equal(expected.length, actual.length, `dataset.${name} should have same number of fields`);
+  actual.forEach((actualField, i) => {
+    cmpField(t, expected[i], actualField, `dataset.${name} fields ${actualField.name}`);
+  });
+}
+
 export function cmpField(t, expected, actual, name) {
-  cmpObjectKeys(t, expected, actual, name);
+  if (expected && actual) {
+    cmpObjectKeys(t, expected, actual, name);
 
-  Object.keys(expected).forEach(k => {
-    if (k === 'filterProps') {
-      // compare filterProps
-      t.ok(typeof actual[k] === 'object', `${name} should have filterProps`);
+    Object.keys(expected).forEach(k => {
+      if (k === 'filterProps') {
+        // compare filterProps
+        t.ok(typeof actual[k] === 'object', `${name} should have filterProps`);
 
-      if (actual[k]) {
-        cmpObjectKeys(t, expected[k], actual[k], `${name} filterProps`);
-        // compare filterProps key
-        Object.keys(expected[k]).forEach(key => {
-          if (key === 'histogram' || key === 'enlargedHistogram') {
-            t.ok(actual[k][key].length, `${name}.filterProps.${key} should not be empty`);
-          } else {
-            t.deepEqual(
-              actual[k][key],
-              expected[k][key],
-              `${name}.filterProps.${key} should be correct`
-            );
-          }
-        });
+        if (actual[k]) {
+          cmpObjectKeys(t, expected[k], actual[k], `${name} filterProps`);
+          // compare filterProps key
+          Object.keys(expected[k]).forEach(key => {
+            if (key === 'histogram' || key === 'enlargedHistogram') {
+              t.ok(actual[k][key].length, `${name}.filterProps.${key} should not be empty`);
+            } else {
+              t.deepEqual(
+                actual[k][key],
+                expected[k][key],
+                `${name}.filterProps.${key} should be correct`
+              );
+            }
+          });
+        }
+      } else if (k === 'valueAccessor') {
+        t.ok(typeof actual[k] === 'function', `${name}.valueAccessor should be a function`);
```

Comment:
we should actually test `valueAccessor` not that it's just a function

Reply:
got it. it's not easy to test equality (not memory address but value) among functions, how should we test it?
I was thinking about the following options:
- calling the function with some data and check that both function return the same value? this may have other implications
- use toString on both functions and compare the output?

Reply:
ok i guess it's fine. the result of valueAccessor is indirectly tested in format layer data and filter.

Reply:
cool